### PR TITLE
Add ThumbprintString property to mgmt

### LINKED ISSUE #33804: The BinaryData format property can't handle byte[] data properly.
image

### PR DESCRIPTION
**Pull request by live1206**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/33804
Add `ThumbprintString` to return the hexadecimal string representation of the SHA-1 hash of the certificate, `Thumbprint` is hidden but still available.
The affected RPs are:
- websites
- automation
- batch
- iothub
- notificationhubs

### CODE REVIEW

File: sdk/automation/Azure.ResourceManager.Automation/src/Customized/AutomationCertificateData.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("{\"key\": \"value\"}")</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// </list>
+        /// </para>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (ArthurMa1978) [inline sdk/automation/Azure.ResourceManager.Automation/src/Customized/AutomationCertificateData.cs:41]: I think we can mark it as obsoleted.

File: sdk/automation/Azure.ResourceManager.Automation/src/Customized/AutomationCertificateData.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        public BinaryData Thumbprint { get; }
```
Comment (ArthurMa1978) [inline sdk/automation/Azure.ResourceManager.Automation/src/Customized/AutomationCertificateData.cs:42]: To make it backward compatible, the `get` should return a BinaryData from `ThumbprintString`.

Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Customized/AutomationCertificateData.cs:42]: And I think we can also mark this property as obsoleted .

File: sdk/automation/Azure.ResourceManager.Automation/src/Customized/Models/AutomationCertificateCreateOrUpdateContent.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("{\"key\": \"value\"}")</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// </list>
+        /// </para>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public BinaryData Thumbprint { get; set; }
```
Comment (ArthurMa1978) [inline sdk/automation/Azure.ResourceManager.Automation/src/Customized/Models/AutomationCertificateCreateOrUpdateContent.cs:42]: To make it backward compatible, the `get` should return a `BinaryData` from `ThumbprintString`, and `set` is update the value of `ThumbprintString`.

Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Customized/Models/AutomationCertificateCreateOrUpdateContent.cs:42]: To make it backward compatible, the get should return a BinaryData from ThumbprintString.

File: sdk/automation/Azure.ResourceManager.Automation/src/Customized/AutomationCertificateData.cs (hunk 2 above)
Comment (live1206) [reply sdk/automation/Azure.ResourceManager.Automation/src/Customized/AutomationCertificateData.cs]: Updated

File: sdk/automation/Azure.ResourceManager.Automation/src/Customized/Models/AutomationCertificateCreateOrUpdateContent.cs (hunk 3 above)
Comment (live1206) [reply sdk/automation/Azure.ResourceManager.Automation/src/Customized/Models/AutomationCertificateCreateOrUpdateContent.cs]: Updated

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**